xdsl/dialects/experimental/stencil.py

```diff
@@ -0,0 +1,369 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import cast, Any
+
+from xdsl.dialects.builtin import (ParametrizedAttribute, ArrayAttr, f32, f64,
+                                   IntegerType, IndexType, IntAttr, AnyFloat)
+from xdsl.ir import Operation, Dialect, MLIRType
+from xdsl.irdl import (irdl_attr_definition, irdl_op_definition, ParameterDef,
+                       AttrConstraint, Attribute, Region, VerifyException,
+                       AnyOf, Annotated, Operand, OpAttr, OpResult, VarOperand,
+                       VarOpResult, OptOpAttr, AttrSizedOperandSegments)
+
+
+@dataclass
+class IntOrUnknown(AttrConstraint):
```

Comment:
Can you describe what this class models? The fact that a given Array has a specific length? Why the name then?

Reply:
Wait isn't this the same as `ArrayLenght` below?